src/runtime/particle-execution-context.ts

```diff
@@ -201,7 +202,7 @@ export class ParticleExecutionContext {
 
   // tslint:disable-next-line: no-any
   async _instantiateParticle(id: string, spec: ParticleSpec, proxies: ReadonlyMap<string, StorageProxy>): Promise<[any, () => Promise<void>]> {
-    let resolve : () => void = null;
+    let resolve: Runnable = null;
```

Comment:
This should be a `Runnable | null`. 

Actually, now that we are mentioning it: should we add an `Optional<T>` type? Or just prefer simple union types? 

`Optional<T>` could be defined as follows: 
```
export type Optional<T> = null | undefined | T;
```

Reply:
On second thought... the `?` operator kind of covers this. 

Instead, should let's mark this as `let resolve: Runnable?`.

Reply:
Null is not the same as undefined.

In this case the variable just not initialized, so I'd remove nullability and just use the undefined syntax.